stake-pool/program/src/processor.rs

```diff
@@ -2224,6 +2410,36 @@ impl Processor {
         Ok(())
     }
 
+    /// Processes [SetStakeDepositAuthority/SetSolDepositAuthority](enum.Instruction.html).
+    fn process_set_deposit_authority(
+        program_id: &Pubkey,
+        accounts: &[AccountInfo],
+        for_stake_deposit: bool,
+    ) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let stake_pool_info = next_account_info(account_info_iter)?;
+        let manager_info = next_account_info(account_info_iter)?;
+
+        let new_sol_deposit_authority = next_account_info(account_info_iter).ok().map(
+            |new_sol_deposit_authority_account_info| *new_sol_deposit_authority_account_info.key,
+        );
+
+        check_account_owner(stake_pool_info, program_id)?;
+        let mut stake_pool = try_from_slice_unchecked::<StakePool>(&stake_pool_info.data.borrow())?;
+        if !stake_pool.is_valid() {
+            return Err(StakePoolError::InvalidState.into());
+        }
+        stake_pool.check_manager(manager_info)?;
+        if for_stake_deposit {
+            stake_pool.stake_deposit_authority = new_sol_deposit_authority.unwrap_or(
+                find_deposit_authority_program_address(&crate::id(), stake_pool_info.key).0,
```

Comment:
Use the passed `program_id` here instead of `crate::id()`, and you won't have to change the crate id